source/common/router/router.cc

```diff
@@ -261,6 +274,9 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   ASSERT(headers.Host());
   ASSERT(headers.Path());
 
+  const Http::HeaderEntry* content_type = headers.ContentType();
+  grpc_request_ = content_type != nullptr &&
+                  Http::Headers::get().ContentTypeValues.Grpc == content_type->value().c_str();
```

Comment:
There are multiple gRPC content types, so I don't think this is sufficient. I think this might actually need to be a prefix match?